### PR TITLE
argmin-negation-overflow: fix aten.min.dim indices at dtype-extremal minima

### PR DESCRIPTION
**Pull request by arames**
## Symptom
`aten.min.dim` returns the wrong `indices` when the true minimum is a
dtype-extremal integer value (e.g. uint8 min `0`, int8 min `-128`). `values`
is correct; only `indices` is wrong. Silent — no crash.

## Root cause
`replace_min_dim` in `coreai_torch/_aten_to_core.py` computed argmin as
`argmax(x * -1)`. Negation via multiply is not order-reversing at integer
extremes: `0 * -1 == 0` (not the range top) for unsigned, and `-128 * -1`
overflows int8. So `argmax` scores the true minimum incorrectly.

## Fix
Reverse order for integer dtypes with the bitwise complement `~x = x ^ -1`
(`broadcasting_bitwise_xor` with an all-ones constant) instead of negation.
`~x = -x - 1` is a strictly decreasing bijection over the full range of every
integer dtype (signed and unsigned), never overflows, and introduces no new
ties, so `argmax(~x) == argmin(x)` with first-index-on-tie semantics preserved.
The float path keeps `x * -1` (exact, no overflow).

## Tests
Adds dtype-extremal coverage (uint8 min 0; int8/int16/int32 dtype-min),
duplicate-minima tie-break, and a float case proving the float path is
unchanged. Reverting the fix fails all integer cases.

### CODE REVIEW

File: tests/ops/test_ops.py [hunk 1]
```diff
@@ -3003,6 +3003,106 @@ def forward(self, x: Tensor) -> tuple[Tensor, Tensor]:
     await validate_numerical_output(model=model, x=x, dynamic_shapes=dynamic_shapes)
 
 
+class TestMinDimArgminDtypeExtremes:
```
Comment (arames) [inline tests/ops/test_ops.py:3006]: @gokulkrishna98 : Should we split this test file per op? Maybe in a separate commit?

File: coreai_torch/_aten_to_core.py [hunk 2]
```diff
@@ -2303,19 +2303,35 @@ def replace_min_dim(
 ) -> OpResultList:
     """Computes min along a specific dimension, returning (values, indices).
 
-    Argmin is computed as argmax(-x), since Core AI has argmax but not argmin.
+    Core AI has argmax but not argmin, so argmin is computed as argmax of an
```
Comment (gokulkrishna98) [inline coreai_torch/_aten_to_core.py]: nit: it would be good for the doc string of op conversion be minimal and terse.

File: tests/ops/test_ops.py (hunk 1 above)
Comment (gokulkrishna98) [reply tests/ops/test_ops.py:3006]: It will create too many files around 160+. Not a hard opinion, we can change it if it gets unmaintainable.

**Review by gokulkrishna98 — APPROVED** [review]

**Comment by gokulkrishna98** [comment]
Please install pre-commit, to resolve linting issues: https://github.com/apple/coreai-torch/blob/main/.pre-commit-config.yaml

File: tests/ops/test_ops.py [hunk 3]
```diff
@@ -3003,6 +3003,106 @@ def forward(self, x: Tensor) -> tuple[Tensor, Tensor]:
...
+        # a future reduce_min-on-bool change is noticed here; if reduce_min
+        # gains i1 support, this test should flip to validate_numerical_output.
+        x = torch.tensor([True, False, True, False])
+        assert x.dtype == torch.bool
+        # First False is the minimum; torch returns its first index.
+        assert int(torch.min(x, dim=0).indices) == 1
+        model = self.MinDimModel(dim=0, keepdim=keepdim).eval()
+        # coreai.reduce_min's MLIR verifier rejects the i1 operand; save_asset
+        # surfaces it as a RuntimeError wrapping the underlying MLIRError.
+        with pytest.raises(RuntimeError):
```
Comment (gokulkrishna98) [inline tests/ops/test_ops.py]: nit: Is it required to test for runtime errors ? it would be good to remove it or skip it.

File: tests/ops/test_ops.py (hunk 1 above)
Comment (arames) [reply tests/ops/test_ops.py:3006]: Multiple files would be much better structure than a single file.
The single file will grow to (more) thousands of lines, and effectively become an unstructured bunch of tests. Splitting and categorizing would be more useful for humans and more efficient for genai.

This can be done separately.

File: coreai_torch/_aten_to_core.py (hunk 2 above)
Comment (arames) [reply coreai_torch/_aten_to_core.py]: Reverted the docstring to the original terse wording, and moved the bijection/order-reversing rationale into an inline code comment instead (on the `~x` branch) rather than the docstring.

File: tests/ops/test_ops.py (hunk 3 above)
Comment (arames) [reply tests/ops/test_ops.py]: Removed `test_bool_input_reduce_min_unsupported` — it was asserting an unrelated, pre-existing limitation (bool input rejected by `coreai.reduce_min` itself, true on `main` too), not behavior introduced by this fix.

**Comment by arames** [comment]
Installed pre-commit and re-ran it over the changed files (`ruff check`, `ruff format`, plus the generic hooks and license-header check) — all clean now. Squashed the branch down to a single commit as well.

**Comment by gokulkrishna98** [comment]
Hi Alex, for merging, your commits must have verified signature. I think you have to setup signing-key ssh for this GitHub account.
